Pull in upstream

CodeWithBryan opened the pull request.

abextm committed:
runelite-client: use FlatLaf

abextm committed:
devtools: add swing inspector

Nightfirecat committed:
timers: Track god wars altar cooldown via varbit

abextm committed:
clientui: optimize for FlatLaf

GriffinBoris committed:
add fishing

GriffinBoris committed:
add fishing

GriffinBoris committed:
Merge remote-tracking branch 'origin/development' into griffin/mining-trainer

GriffinBoris committed:
save point for woodcutting

GriffinBoris committed:
Merge remote-tracking branch 'origin/development' into griffin/mining-trainer

GriffinBoris committed:
add woodcutting and fix bugs

GriffinBoris committed:
add equip bool to item helper

GriffinBoris committed:
Merge remote-tracking branch 'origin/development' into griffin/mining-trainer

GriffinBoris committed:
Add antiban and auto pilot stuff

GriffinBoris committed:
Add antiban and auto pilot stuff

GriffinBoris committed:
Merge remote-tracking branch 'origin/development' into griffin/mining-trainer

abextm committed:
config: optimize for FlatLaf

Adam- committed:
loottracker: include relic selections in npc metadata

DetectiveCode committed:
Improved the agility script.
- Added random pausing which can be turned on or off via the configuration.
- Added support for eating food and configuring HP threshold to eat food at.
- Script now stops when the HP threshold is above 0 and there's no food in the inventory.
- Added support for Pollnivneach rooftop agility course.

DetectiveCode committed:
Improved and rewrote the summers garden script.
- Added functionality to automatically hand in the "Summer sq'irkjuice" to Osman.
- Added functionality to automatically get beer glasses from the shelves.
- Added configuration options to disable checking for elemental collision for mass worlds.
- Added configuration options for full automation and semi-automation via "Auto Maze Completion" and "Auto Hand in and Reset".
- Rewrote the guide to provide more details on how to use the adjusted script.

DetectiveCode committed:
Adjusted some config names and tags.

GriffinBoris committed:
Merge remote-tracking branch 'origin/development' into griffin/mining-trainer

GriffinBoris committed:
fix for get item

GriffinBoris committed:
fix for get item

chsami committed:
Merge pull request #157 from DetectiveCode/dc/cleanup

Adjusted some config names and tags.

chsami committed:
Merge pull request #156 from DetectiveCode/dc/summers-garden-improvement

Improved and rewrote the summers garden script.

chsami committed:
Merge pull request #155 from DetectiveCode/dc/agility-improvement

Improved the agility script.

chsami committed:
Merge pull request #160 from GriffinBoris/griffin/fix-get-item-call

fix for get item

geheur committed:
Trailblazer league reloaded rune pouch up to 6 runes.

Signed-off-by: sam <dasistkeinnamen@gmail.com>

GriffinBoris committed:
Merge remote-tracking branch 'origin/development' into griffin/mining-trainer

cwjoshuak committed:
clues: fix single item fulfilledBy check

The item has no quantity, so testing it against quantity is wrong.
This fixes testClueTag() matching MultipleOfItemRequirements when
the quantity is >1.

GriffinBoris committed:
save point

Adam- committed:
containable frame: remove setMaximizedBounds workaround

Flatlaf uses the native ShowWindow method to maximize the screen, making the setMaximizedBounds code unused on Windows.

abextm committed:
api: add Animation::restartMode accessors

GriffinBoris committed:
save point

GriffinBoris committed:
remove changes

Commit:
Update Item variations to 2023-11-15

Commit:
Update Scripts to 2023-11-15

Commit:
Update Widget IDs to 2023-11-15

Commit:
Update Item IDs to 2023-11-15

Commit:
Update NPC IDs to 2023-11-15

Commit:
Update Object IDs to 2023-11-15

Commit:
Release 1.10.17

Commit:
Bump for 1.10.18-SNAPSHOT

[ci skip]

Adam- committed:
null check getSelectedWidget() on menu click

abextm committed:
runepouch: skip empty slots in grid view

Adam- committed:
gpu: regions: add pyramid plunder

Adam- committed:
prayer: move drain rate to prayer plugin

Adam- committed:
prayer: add drain rate test

Adam- committed:
prayer: convert drain rate to drain effect

Adam- committed:
prayer: add ruinous powers

varbit  varp lower   upper
14827   3780    0       23
14829   3780    0       0
14830   3780    1       1
14831   3780    2       2
14832   3780    3       3
14833   3780    4       4
14834   3780    5       5
14835   3780    6       6
14839   3780    10      10
14840   3780    11      11
14841   3780    12      12
14842   3780    13      13
14843   3780    14      14
14844   3780    15      15
14845   3780    16      16
14846   3780    17      17
14847   3780    18      18
14848   3780    19      19
14850   3780    21      21
14851   3780    22      22
14962   3780    7       7
14963   3780    8       8
14964   3780    9       9
14965   3780    20      20
14966   3780    23      23

prayer bit
Ancient Strength 0
Ancient Sight 1
Ancient Will 2
Trinitas 3
Decimate 4
Annihilate 5
Vaporise 6
Dampen Melee 7
Dampen Ranged 8
Dampen Magic 9
Purge 10
Rejuvenation 11
Ruinous Grace 12
Wrath 13
Metabolise 14
Berserker 15
Fumus' Vow 16
Cruor's Vow 17
Umbra's Vow 18
Glacies' Vow 19
Intensify 20
Rebuke 21
Vindication 22
Protect Item 23

Adam- committed:
xpdrop: support multiple prayer types per prayer

Adam- committed:
xpdrop: rename package to xpdrop

Adam- committed:
xpdrop: add ruinous powers

GriffinBoris committed:
changes

abextm committed:
api: expose drawFrustum & drawWidgetText

Adam- committed:
rs: add js5connect error message

Game updates now include 30 minutes of downtime a week, causing the js5connect error because the client is loaded from the backup config

Felanbird committed:
achievement diary: update fremennik astral rune task (#17221)

Adam- committed:
clues: add treasure chest bank tag

Macweese committed:
clues: add felling axes

Macweese committed:
clues: add trailblazer tools

Nightfirecat committed:
loot tracker: Track unsired loot

Macweese committed:
clues: fix comment grammar

YvesW committed:
prayer: disable reordering on shutdown

chsami committed:
Merge pull request #161 from GriffinBoris/griffin/mining-trainer

Griffin/mining trainer

Commit:
Update Item variations to 2023-11-29-rev218

Commit:
Update Object IDs to 2023-11-29-rev218

Commit:
Update Item IDs to 2023-11-29-rev218

Adam- committed:
Bump to 1.10.17.1-SNAPSHOT

Adam- committed:
Merge branch 'master' into 1.10.17

Commit:
Release 1.10.17.1

Commit:
Bump for 1.10.17.2-SNAPSHOT

[ci skip]

Adam- committed:
bank tags: refactor ui to use layers

This rewrites most of the tab interface code to use a layer for both the
tab list and also the tag tab tab. The scrolling has been reimplemented
to use native scrolling.

Also modify the bank searching script hook to check items even when the
bank is not in searching mode, which removes some complexity with having
to have the bank in a pseudo-search mode but still allow other searches
to be started.

Replace the import/export notifications with chat messages, which are
always actually delivered and aren't annoying.

Adam- committed:
api: add widget createStaticChild

Adam- committed:
api: add widget setOnScrollWheelListener

Adam- committed:
xp drops: fix coloring defensive casting

Adam- committed:
clues: fix treasure chest tag test

Groups with more than one item were only testing the first item

Adam- committed:
bank tags: rebuild tag tab tab on new/import/delete

Adam- committed:
bank tags: fix scrolling the tag tab

Having a separate layer for the tag tab tab would require either a
separate scrollbar or adjusting the bank scrollbar dynamically between
the two layers.

This is somewhat tricky, so instead just use the previous behavior of
creating the tag tab tab items after the other bank components (this is
so they dont interfere with the normal bank items which have their
clickmask set to send ops).

Move the rebuildTagTabTab() call that was running post bankmain_build to
be instead pre bankmain_finish so that it runs prior to the scrollbar
calculation.

Adam- committed:
bank tags: remove separators prior to computing scroll

Adam- committed:
bank tags: remove use of dummy script null

script null predates the creation of JavaScriptCallback, and was used to attach dummy callbacks to components so that the ops could later be handled. This is no longer used.

Adam- committed:
menu swapper: fix ui swapping optarget

Adam- committed:
bank tags: fix pmd violation

Adam- committed:
raids: simplify login scout check

This also fixes the plugin sending spurious RaidReset events whenever a
map load is hit.

Adam- committed:
raids: fix scouting on raid reload

Reload sets party=-1 and then raid=0, so just reset if raid=0 happens
when not in a party.

Adam- committed:
Merge branch 'master' into 1.10.17

Commit:
Release 1.10.17.2

Commit:
Bump for 1.10.17.3-SNAPSHOT

[ci skip]

chsami committed:
Merge remote-tracking branch 'runelite/1.10.17' into development

# Conflicts:
#	cache/pom.xml
#	pom.xml
#	runelite-api/pom.xml
#	runelite-client/pom.xml
#	runelite-client/src/main/java/net/runelite/client/plugins/timetracking/farming/CompostTracker.java
#	runelite-jshell/pom.xml
#	runelite-maven-plugin/pom.xml

chsami committed:
Merge remote-tracking branch 'origin/development' into development

chsami committed:
1.10.17.2

Adam- committed:
chat filter: cache filter results

The canFilterPlayer check can't be easily cached since it would have to
be invalidated whenever the fc/clan/friends list changes.

chsami committed:
updated to 1.10.17.2

chsami committed:
Merge pull request #162 from chsami/development

Development

DetectiveCode committed:
Improved the Fletching script
- Fixed an issue where it would keep withdrawing a knife.
- Fixed an issue where it would close the bank even though it hadn't withdrawn logs or deposited the product.
- A minor bug I noticed is that when you stop the script while it is retrieving logs from the bank, the menu is changed and does not change back. Even clicking on an item will show as "Withdraw 1 Yew Log" for example. At this point you have to restart the client. Even restarting the script doesn't fix it.

dginovker commented:
Heads up, you're more than welcome to use the scripts in :bettermotherload but to avoid cross-analysis antiban I highly recommend tweaking small values (like sleep times) before use

CodeWithBryan commented:
> Heads up, you're more than welcome to use the scripts in :bettermotherload but to avoid cross-analysis antiban I highly recommend tweaking small values (like sleep times) before use

Not using existing scripts, just the improvements from the branch :)

Adam- committed:
raids: reset raid on party id change

This doesn't happen normally. There is a race which allows creating
multiple parties in the same fc in the same world, by resetting the raid
and then on a 2nd account interacting with the board to quickly make a
new raid at the same time.

In this state, when the raid is reset, it resets the party id to the
other party for the fc like this:

partyid = other
in raid = false
in raid = true
partyid = -1
partyid = new

Instead of the usual reset which looks like this:

partyid = -1
in raid = false
in raid = true
partyid = new

So detect changing between party ids and reset the raid then.

Changing from party id -> -1 is valid and happens when a raid starts,
and changing from -1 -> party id happens after in_raid is flipped and
the raid has been scouted.

pajlada committed:
clues: reword Varrock Apothecary cryptic clue solution

pajlada committed:
clues: clarify Burthorpe Games Room emote clue location

geheur committed:
bank tags: only close chatbox interface when clicking bank ops

This restores the previous behavior which had a hidden check here for the bank.

Adam- committed:
bank tags: move onMenuOptionClicked subscriber to TabInterface

Adam- committed:
bank tags: remove unused import

Adam- committed:
api: add scene tile model/paint isFlat()

Adam- committed:
gpu: fix tile uvs

The client passes different local coordinates to the texture rasterizer
depending on if the tile is flat (all corners have the height) or not.

The code we've had this entire time was always assuming flat tiles. This
used to kind of work, but with the recent changes to compute the uvs in
the shader to correct for textris, it has made non-flat tiles more
broken.

Adam- committed:
gpu: flip y/z in projection

Flipping y/z is necessary to go from jagex to gl coordinates. This was
doing it via rotx by over rotating the camera so that it was looking
down at the scene upside down.

This took me a bit to figure out and I think is unintuitive. Flip y/z in
projection() instead. The new matrix is (in row-major order)

2/w      0     0      0
0     -2/h     0      0
0        0     1    -2n
0        0     1      0

Note that, as with the previous behavior, the final z values are [0,1)
instead of [0,-1) like gl uses for NDC. Since we do not use the z values
currently that does not matter.

Nightfirecat committed:
slayer: Fix initial amount when task changes

This is most commonly encountered when players are assigned a Tzhaar
task, but elect to upgrade it to a Jad or Zuk task. When this happens,
the SLAYER_TASK_SIZE and SLAYER_TASK_CREATURE varps are set when
starting the assignment dialog, and are later changed after opting in to
the task change. The plugin had code which would preserve the previous
initial amount (as of 1dbf432377f4b0355117b25ecde66915456b0ff8) but
should no longer be needed as we can update it directly from var changes
as opposed to tracking it ourselves.

Adam- committed:
gpu: reverse z

This changes the projection to (in row-major order):

2/w      0     0      0
0     -2/h     0      0
0        0     0     2n
0        0     1      0

Which changes the z calculation from (z-100)/z to 100/z.

Adam- committed:
clues: include alternatives in treasure chest tag

Adam- committed:
gpu: remove radius from compute shaders

Since we are not using the distances to index arrays, we can just store
negative distances, making the radius unnecessary.

Adam- committed:
gpu: use floating point for vertex positions

Adam- committed:
api: include WidgetConfig in runtime api

It has a method, transmitAction

Felanbird committed:
idle notifier: add cow milking animation id

Felanbird committed:
idle notifier: add dairy churn animation ids

Adam- committed:
music: correctly apply ambient sound mute on startup and shutdown

thecage555 committed:
ground items: fix inferno despawn timer

PortAGuy committed:
party: tie ping duration to system time

Adam- committed:
clues: use builder for cryptic clues

The code for this was dumped via a test:

	@Test
	public void dump() throws IllegalAccessException
	{
		Map<Integer, String> map = new HashMap<>();
		for (Field f : ObjectID.class.getDeclaredFields())
		{
			map.put(f.getInt(null), f.getName());
		}
		for (CrypticClue cc : CrypticClue.CLUES)
		{
			System.out.println("CrypticClue.builder()");
			System.out.println(".text(\"" + cc.getText() + "\")");
			if (!"Viggora".equals(cc.getNpc()))
			{
				WorldPoint wp = cc.getLocation(plugin);
				if (wp != null)
				{
					int plane = wp.getPlane();
					int x = wp.getX();
					int y = wp.getY();
					System.out.println(".location(new WorldPoint(" + x + ", " + y + ", " + plane + "))");
				}
			}
			else
			{
				System.out.println(".locationProvider(CrypticClue::getViggoraLocation)");
			}
			if (cc.getNpc() != null)
			{
				System.out.println(".npc(\"" + cc.getNpc() + "\")");
			}
			if (cc.getObjectId() != -1)
			{
				var name = map.get(cc.getObjectId());
				if (name != null)
				{
					System.out.println(".objectId(ObjectID." + name + ")");
				}
				else
				{
					System.out.println(".objectId(NullObjectID.NULL_" + cc.getObjectId() + ")");
				}
			}
			System.out.println(".solution(\"" + cc.getSolution() + "\")");
			if (cc.isRequiresLight())
			{
				System.out.println(".requiresLight(true)");
			}
			if (cc.getQuestionText() != null)
			{
				System.out.println(".questionText(\"" + cc.getQuestionText() + "\")");
			}
			System.out.println(".build(),");
		}
	}

Adam- committed:
clues: add npc regions for cryptic clues

Co-authored-by: Macweese <50101641+Macweese@users.noreply.github.com>

ldahleen54 committed:
config: allow using tab key for hotkeys

Adam- committed:
fishing: fix tracking Karambwanji

Felanbird committed:
clues: add fairy ring to mudskipper point emote step

Commit:
Release 1.10.18

Commit:
Bump for 1.10.19-SNAPSHOT

[ci skip]

Adam- committed:
gpu: fix cl compilation on amd gpus

error: call to 'mix' is ambiguous
    float h1 = mix(tile_height(tileHeightImage, plane, sx, sz), tile_height(tileHeightImage, plane, sx + 1, sz), fract(fx, &it));
(cherry picked from commit 7935a7615ce3fc6aa668e78416997e3fec88223f)

Adam- committed:
gpu: fix cl compilation on amd gpus

error: call to 'mix' is ambiguous
    float h1 = mix(tile_height(tileHeightImage, plane, sx, sz), tile_height(tileHeightImage, plane, sx + 1, sz), fract(fx, &it));

Adam- committed:
Bump to 1.10.18.1-SNAPSHOT

Commit:
Release 1.10.18.1

Commit:
Bump for 1.10.18.2-SNAPSHOT

[ci skip]

Adam- committed:
Revert "gpu: fix cl compilation on amd gpus"

This reverts commit 7935a7615ce3fc6aa668e78416997e3fec88223f.

Adam- committed:
Revert "gpu: use floating point for vertex positions"

This reverts commit 95a59215ce635d391862205f8108c131410dbeb4.

Adam- committed:
Revert "gpu: fix cl compilation on amd gpus"

This reverts commit 11d59d203e7ed7e535b63ab77e215e1a67f902d4.

Adam- committed:
Revert "gpu: use floating point for vertex positions"

This reverts commit 95a59215ce635d391862205f8108c131410dbeb4.

Commit:
Release 1.10.18.2

Commit:
Bump for 1.10.18.3-SNAPSHOT

[ci skip]

SuperNerdEric committed:
notifier: add flash taskbar option

Adam- committed:
world hopper: cleanup

This moves the initial panel rebuild to the executor, which can wait on
the world result future if it needs to and not block swwing startup.

Adam- committed:
world hopper: add us east/west coast flags

Co-authored-by: burkeg <burkegabriel1145@gmail.com>
Co-authored-by: geheur <41499327+geheur@users.noreply.github.com>

Adam- committed:
time tracking: support payment tracking of multi-patch npcs

Co-authored-by: Erik <eribi813@student.liu.se>

chsami committed:
Merge remote-tracking branch 'runelite/1.10.18' into development

# Conflicts:
#	cache/pom.xml
#	pom.xml
#	runelite-api/pom.xml
#	runelite-client/pom.xml
#	runelite-jshell/pom.xml
#	runelite-maven-plugin/pom.xml

chsami committed:
updated to 1.10.18

chsami committed:
Merge pull request #164 from chsami/development

Development

chsami committed:
Merge pull request #163 from DetectiveCode/dc/fletching-fix

Improved the Fletching script

Adam- committed:
cryptic clues: fix object id initializer

This defaults to 0 not -1, and I don't see a way to use @Builder.Default with a constructor like this. Just assume 0 is invalid.

pkhermouch committed:
cryptic clues: update Sinclair Mansion solution

Use top instead of first floor to avoid confusion between UK and US floor numbering

Jrod7938 committed:
initial commit of Auto Vorkath (STILL BUGGED ASF)

Commit:
Update Widget IDs to 2024-1-10

lost interface [kourend_favour] 246

Commit:
Update Item IDs to 2024-1-10

Commit:
Update Quests to 2024-1-10

Commit:
Update Item variations to 2024-1-10

Commit:
Update NPC IDs to 2024-1-10

Commit:
Update Object IDs to 2024-1-10

Felanbird committed:
idle notifier: add specimen cleaning animation ids

Jrod7938 committed:
updated ProjectileMoved to allow us to subscribe to projectile movements

Jrod7938 committed:
AutoVorkath Complete

Jrod7938 committed:
AutoVorkath Complete

Jrod7938 committed:
AutoVorkath bug fixed

Jrod7938 committed:
AutoVorkath bug fixed

raiyni committed:
overlay manager: revalidate overlays on profile change

WidgetOverlays need to be revalidated so the moved widget can be moved back to its original position.

Signed-off-by: Ron Young <admin@ryoung.io>

Jrod7938 committed:
AutoVorkath bug fixed

Jrod7938 committed:
AutoVorkath bug fixed

Jrod7938 committed:
AutoVorkath bug fixed

Commit:
Update 219

abextm committed:
fairyring: add vinery fairy ring

Commit:
Release 1.10.19

Commit:
Bump for 1.10.20-SNAPSHOT

[ci skip]

Nightfirecat committed:
clues: Fix Lovada cryptic clue text

With today's game update, favour has been removed from Great Kourend,
and also from this clue's text.

Felanbird committed:
achievement diary: add new kourend task

Felanbird committed:
achievement diary: remove kourend favour

Felanbird committed:
clues: update lizardman canyon hot-cold step

Felanbird committed:
world map: add AKR fairy ring

abextm committed:
kourendlibrary: remove dark manuscripts

testing-ongithub committed:
worldmap: fix Lovakengj mine cart location

testing-ongithub committed:
worldmap: add new minecart locations

testing-ongithub committed:
worldmap: rename minecarts to match in-game list

testing-ongithub committed:
worldmap: add ALR fairy ring

testing-ongithub committed:
worldmap: fix DIP fairy ring location

testing-ongithub committed:
worldmap: alphabetize fairy rings

testing-ongithub committed:
worldmap: use Zanaris fairy code rather than location

This affects how its tooltip appears in the World Map

testing-ongithub committed:
worldmap: comment fairy rings that don't exist on the world map

chsami committed:
Merge remote-tracking branch 'runelite/master' into development

# Conflicts:
#	cache/pom.xml
#	pom.xml
#	runelite-api/pom.xml
#	runelite-client/pom.xml
#	runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java
#	runelite-jshell/pom.xml
#	runelite-maven-plugin/pom.xml

chsami committed:
updated to 1.10.20

chsami committed:
Merge pull request #166 from chsami/development

Development

CreativeTechGuy committed:
npc aggro area: ignore unattackable NPCs

smol-tako committed:
item identification: add Forgotten brew and Blighted super restore

chsami committed:
update ci

geheur committed:
prayer: reorder prayers on profile change

Yenof committed:
clues: Add Team Cape 0/X/I to emote clue steps (#17273)

Co-authored-by: Felanbird <41973452+Felanbird@users.noreply.github.com>

chsami committed:
Merge pull request #165 from Jrod7938/main

[JR] Auto Vorkath Complete

Commit:
Release 1.10.20

Commit:
Bump for 1.10.21-SNAPSHOT

[ci skip]

abextm committed:
clientui: apply client size config correctly

Adam- committed:
ui: remove contain in screen ALWAYS

Previously with custom chrome, the window had no title bar and dragging
was supported via the substance root pane using a mouse listener which
called setLocation. Thus restricting the frame to a single screen was
possible with overring setLocation.

Now we have a native window with a native titlebar, dragging is
performed natively, and the location is not set via setLocation(). We
had attempted to work around this by registering a component moved
listener, but it runs after the window is actually moved and the end
result is we end up fighting with the native drag system, which is
visibily jarring.

It looks like it might be possible to do this via a custom wndproc
checking for SC_MOVE, but it would probably require some extensive
flatlaf changes and also is not portable. Remove the option instead.

Adam- committed:
IconTextField: fix popup suggestion list size

Adam- committed:
config: use default font for JPasswordField

The default font for this is now applied by flatlaf to be the rs font. Set it back to Dialog font so it doesn't rely on font fallback anymore. See ca56ef10823aea1ecd275c0d3c3f9a3b17838ff9

Adam- committed:
containable frame: fix Window.setMinimumSize DPI scaling

Fixes JDK-8221452 - Window.setMinimumSize does not respect DPI scaling

This has never worked previously with a native titlebar, but the
Substance custom chrome was handling it correctly, and so it was
unnoticed.

abextm committed:
clientui: use sidebar pref width as min width

Layout uses the sidebar's pref width as its actual width always, so it
is basically both its min and max width. Panels that are wrapped don't
have a min size set, so the whole window's min size was smaller than
Layout would let it me, causing it to fight and sometimes loose.

abextm committed:
clientui: add 4px border in custom chrome mode

A lot of users were very used to having this from substance

abextm committed:
clientui: do not try to shift the frame in screen when not changing size

Adam- committed:
laf: disable text antialiasing for rs fonts

This fixes cleartype mangling the text color, making it visibly different from substance

Adam- committed:
banktags: fix tabtabs tab

This was inadvertently broken previously.

Allows making a named tagtab named tagtabs to open the tag tab view.

Nightfirecat committed:
clientui: Support Linux WM layout

On Linux, FlatLaf will support both yielding window layout to the window
manager (for maximizing, edge snapping, etc.) as well as displaying a
decorated frame or dialog window only if the following are set:

```
JFrame.setDefaultLookAndFeelDecorated(true);
JDialog.setDefaultLookAndFeelDecorated(true);
```

While these properties are usually already covered by our setting the
client frame to be undecorated and the root panel to be decorated as a
frame, [FlatLaf explicitly checks these properties][1], so they must be
set to take effect in Linux. For reference, [FlatLaf's demo program also
sets these properties][2] and also creates a window which can be
layouted by the window manager.

[1]: https://github.com/JFormDesigner/FlatLaf/blob/3.2.5/flatlaf-core/src/main/java/com/formdev/flatlaf/ui/FlatNativeLinuxLibrary.java#L101-L104
[2]:https://github.com/JFormDesigner/FlatLaf/blob/3.3/flatlaf-demo/src/main/java/com/formdev/flatlaf/demo/FlatLafDemo.java#L63-L68

abextm committed:
clientui: do not show custom chrome border when maximized

abextm committed:
clientui: do not include insets in game size config

abextm committed:
clientui: correctly apply suction when expanding via min size change

when expanding the sidebar while 'keep game size' was set, and the game
width was less than the min size of the window after expanding,
rightSideSuction would not be set. This is because setting the frame's
min size causes the frame to immediately resize if it is too small, so
ContainableFrame would not see a simultaneous expand + shift like it
expects.

Adam- committed:
worldpoint: use signed right shift for local to world conversion

Adam- committed:
laf: change titlepane height to 27px

titlepane height is set by the buttons, which default to 44,30

Substance titlebar is 27px, so use 27px. 33 was picked because it is a little bigger than 27.

abextm committed:
ContainableFrame: fix dpi scaled minimum size hack

abextm committed:
ContainableFrame: restore old rhs offset behavior

Adam- committed:
screenshot: image capture: support insets on client frame

The insets include the native client border, which results in an 8px
alpha border on the image. Remove the insets from the screenshot.

With j2d scaling the image size is an extra pixel wide and the client
position within the frame is an extra pixel down (with gpu). I think
this is due to the rounding of getScaledValue(), which was originally
copied from jogl.

Adam- committed:
ui: readd contain in screen ALWAYS

This is now implemented in flatlaf's native window border library

Adam- committed:
ui: remove window opacity setting

This only works with undecorated frames, and also didn't work on Linux.

Now that we never use undecorated frames anymore on Windows, this setting has no effect.

Adam- committed:
update flatlaf to 3.2.5-rl2

Commit:
Update Object IDs to 2024-01-24-rev219

Commit:
Update Item IDs to 2024-01-24-rev219

Commit:
Update Item variations to 2024-01-24-rev219

Commit:
Update NPC IDs to 2024-01-24-rev219

Adam- committed:
add scurrius

Adam- committed:
hiscores: add scurrius boss icon

Nightfirecat committed:
clues: Add Wizards' tower region to Wizards' tower clue

Adam- committed:
devtools: cleanup location overlays

Adam- committed:
ContainableFrame: apply insets to old rhs offset behavior

Adam- committed:
laf: change titlepane height to 23px

Substance titlebar is 23px with a 4px border. Set titlebar height to 23px and then add a 4px border. This fixes the titlebar to be 23px when maximized vertically instead of 27px.

Adam- committed:
clientui: adjust for frame insets when performing display bounds check

This prevents the client resetting its position when it is too close to the display edge due to the insets

abextm committed:
ContainableFrame: set suction when shifted by native containment

abextm committed:
ContainableFrame: use hungarian to prevent window / content coord misuse

Adam- committed:
ui: readd window opacity setting

The undecorated frame requirement appears to not be real, at least on Windows.

AWT applies translucency via setting WS_EX_LAYERED to make the window layered and then changes the opacity of the layered window with SetLayeredWindowAttributes.

This appears to work regardless of the frame state, even with custom chrome off.

Adam- committed:
use backgroundless logo for image icon

Also add a 16px pixelart logo for the title bar and tray icon

Limero committed:
xp updater: add runetracker support

abextm committed:
clientui: synchronously update the root pane's size

Previously it was possible for the first layout of a panel to cause a
second layout that could observe the size of the root pane before it was
updated with the new size of the frame from the first layout of
ClientUI, causing it to shrink the client when it is not supposed to.

abextm committed:
clientui: listen for sidebar hotkeys when the client is not focused

abextm committed:
clientui: give client focus when closing or hiding the sidebar

Adam- committed:
update flatlaf to 3.2.5-rl3

Adam- committed:
Revert "update flatlaf to 3.2.5-rl3"

This reverts commit b3cef61d63b76d0dcfe68f744c7e307914f0d31a.

Commit:
Update Scripts to 2024-1-31

Commit:
Update Widget IDs to 2024-1-31

Commit:
item variations: do not merge full helms with other helms

The crystal 'full' items are all unobtainable, so do not matter.

abextm committed:
rl-client: remove ItemVariationMappingTest

This mostly just tests that the json file hasn't changed, which is not
useful

Commit:
Release 1.10.21

Commit:
Bump for 1.10.22-SNAPSHOT

[ci skip]

Adam- committed:
update flatlaf to 3.2.5-rl4

JZomDev committed:
ui: don't apply opacity in safe mode

Felanbird committed:
worldmap: fix Kourend teleport level

iProdigy committed:
chat commands: fix off-by-one team size for out-of-order toa pb

chsami committed:
Merge branch 'master' into development

# Conflicts:
#	pom.xml
#	runelite-client/src/main/java/net/runelite/client/plugins/devtools/LocationOverlay.java
#	runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java
#	runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java
#	runelite-client/src/main/java/net/runelite/client/ui/SplashScreen.java

chsami committed:
Runelite update 1.10.22

chsami committed:
fix inventorysetupPlugins

chsami committed:
remove javadoc

chsami committed:
Merge pull request #167 from chsami/development

Development

chsami committed:
fix project.version

chsami committed:
Merge pull request #168 from chsami/development

fix project.version

chsami committed:
fix special attack & overloads nmz script

chsami committed:
Merge pull request #169 from chsami/development

fix special attack & overloads nmz script
